Resolved revoke ballot issue & added security fixes

007vedant opened the pull request:
- implemented passcode based encryption and decryption of ballots using PyNaCl
- modified the casting vote workflow to resolve issue of revoke ballot being CPU intensive
- modified the db schema to support the new workflow and encryption

jberkus commented:
New code looks OK, am going to test it out.

kalkayan commented on elekto/controllers/elections.py:
```diff
@@ -94,120 +101,158 @@ def elections_voting_page(eid):
+                enc_ballot_ids.append(b64encode(enc_ballot_id).decode("utf-8"))
                 e.ballots.append(ballot)
 
+        voter.ballot_ids = json.dumps(
+            enc_ballot_ids
+        )  # json serialized encrypted ballot ids
```
Aren't we storing a UUID for the voter in the `ballot.voter` of all the votes and storing the encryption of that UUID in `voter.encrypted_ballot` instead of `voter.ballot_ids`?

007vedant replied:
@kalkayan I'll do the required modifications as per our discussion.

007vedant replied:
@kalkayan modifications done as per the review via latest updates

007vedant commented:
- performed encryption of `ballot.voter` (UUID) as per the latest review
- updated `voter.ballot_id` to LargeBinary type. (stores encrypted byte string of `ballot.voter`.
- added `encryption.py` module in `elekto/middlewares` to provide encryption/decryption APIs.
- removed redundant code.

jberkus commented:
You need to add pynacl to requirements.txt

007vedant commented:
- modified UUID type which now force convert UUID to stringified hex values for postgres storage
- added latest stable version of PyNaCl to requirements.txt

jberkus commented:
@kalkayan, wanna give your +1 here before we merge?

kalkayan commented:
We can merge this @jberkus.
